x-pack/platform/plugins/shared/triggers_actions_ui/public/application/components/rules_setting/rules_settings_flyout.test.tsx

```diff
@@ -0,0 +1,454 @@
+};
+
+const RulesSettingsFlyoutWithProviders: React.FunctionComponent<RulesSettingsFlyoutProps> = (
+  props
+) => (
+  <IntlProvider locale="en">
+    <QueryClientProvider client={queryClient}>
+      <RulesSettingsFlyout {...props} />
+    </QueryClientProvider>
+  </IntlProvider>
+);
+
+const waitForFlyoutLoad = async (options?: {
+  flappingSection?: boolean;
+  queryDelaySection?: boolean;
+}) => {
+  await waitFor(() => {
+    expect(screen.queryByTestId('rulesSettingsFlyout')).not.toBe(null);
+  });
+
+  const { flappingSection = true, queryDelaySection = true } = options || {};
+
+  await waitFor(() => {
+    if (flappingSection) {
+      expect(screen.queryByTestId('rulesSettingsFlappingSection')).toBeInTheDocument();
+    }
+    if (queryDelaySection) {
+      expect(screen.queryByTestId('rulesSettingsQueryDelaySection')).toBeInTheDocument();
+    }
+  });
+};
+
+describe('rules_settings_flyout', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+
+    const [
+      {
+        application: { capabilities },
+      },
+    ] = await mocks.getStartServices();
+    useKibanaMock().services.application.capabilities = {
+      ...capabilities,
+      rulesSettings: {
+        save: true,
+        show: true,
+        writeFlappingSettingsUI: true,
+        readFlappingSettingsUI: true,
+        writeQueryDelaySettingsUI: true,
+        readQueryDelaySettingsUI: true,
+      },
+    };
+
+    useKibanaMock().services.notifications.toasts = {
+      addSuccess: jest.fn(),
+      addError: jest.fn(),
+      addDanger: jest.fn(),
+      addWarning: jest.fn(),
+    } as unknown as IToasts;
+
+    useKibanaMock().services.isServerless = true;
+
+    fetchFlappingSettingsMock.mockResolvedValue(mockFlappingSetting);
+    updateFlappingSettingsMock.mockResolvedValue(mockFlappingSetting);
+    getQueryDelaySettingsMock.mockResolvedValue(mockQueryDelaySetting);
+    updateQueryDelaySettingsMock.mockResolvedValue(mockQueryDelaySetting);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    queryClient.clear();
+    cleanup();
+  });
+
+  test('renders flapping settings correctly', async () => {
+    const result = render(<RulesSettingsFlyoutWithProviders {...flyoutProps} />);
+    expect(fetchFlappingSettingsMock).toHaveBeenCalledTimes(1);
+    await waitForFlyoutLoad();
+    expect(
+      result.getByTestId('rulesSettingsFlappingEnableSwitch').getAttribute('aria-checked')
+    ).toBe('true');
+    expect(result.getByTestId('lookBackWindowRangeInput').getAttribute('value')).toBe('10');
+    expect(result.getByTestId('statusChangeThresholdRangeInput').getAttribute('value')).toBe('10');
+
+    expect(result.getByTestId('rulesSettingsFlyoutCancelButton')).toBeInTheDocument();
+    expect(
+      result.getByTestId('rulesSettingsFlyoutSaveButton').getAttribute('disabled')
+    ).toBeFalsy();
+  });
+
+  test('can save flapping settings', async () => {
+    const result = render(<RulesSettingsFlyoutWithProviders {...flyoutProps} />);
```

Comment:
```suggestion
    render(<RulesSettingsFlyoutWithProviders {...flyoutProps} />);
```

nit: `screen` is preferred for querying

Reply:
Sorry, somehow I created a copy of the this file before renaming it and github thought this was a new file. Regarding using `screen`, as there is not just this test that would need to be updated I prefer to do so in another PR dedicated to that and not mix things up